crates/rslint_core/src/groups/errors/no_new_symbol.rs

```diff
@@ -0,0 +1,72 @@
+use crate::rule_prelude::*;
+use ast::*;
+use SyntaxKind::*;
+
+declare_lint! {
+    /**
+    Disallow constructing `Symbol` using `new`.
+
+    `Symbol` shouldn't be constructed using `new` keyword since it results in a `TypeError`, instead
+    it should be called as a function.
+
+    ## Incorrect code examples
+
+    ```js
+    // This call results in TypeError
+    const fooSymbol = new Symbol("foo"); 
+    ```
+
+    ## Correct code examples
+
+    ```js
+    const fooSymbol = Symbol("foo");
+    ```
+    */
+    #[derive(Default)]
+    NoNewSymbol,
+    errors,
+    "no-new-symbol",
+}
+
+#[typetag::serde]
+impl CstRule for NoNewSymbol {
+    fn check_node(&self, node: &SyntaxNode, ctx: &mut RuleCtx) -> Option<()> {
+        if node.kind() == NEW_EXPR
+        {
+            let new_expr = node.to::<NewExpr>();
+
+            if new_expr.object()?.syntax().text() == "Symbol" {
+                let err = ctx
+                    .err(
+                        self.name(),
+                        "`Symbol` cannot be called as a constructor.",
+                    ).primary(
+                        new_expr.new_token().unwrap().text_range(),
+                        "this operator is redundant...",
```

Comment:
```suggestion
                    ).primary(
                        node,
                        "",
```
That error message is a bit inconsistent, we use `...` to link label logic together but that does not link to anything else, also, the range should be the whole invocation